File: 344.反转字符串.go
/*
 * @lc app=leetcode.cn id=344 lang=golang
 *
 * [344] 反转字符串
 */

// @lc code=start
func reverseString(s []byte)  {
	for left, right := 0, len(s) - 1; left < right; left, right = left+1, right-1 {
		s[left] ^= s[right]
		s[right] ^= s[left]
		s[left] ^= s[right]
	}
}
// @lc code=end

File: leetcode/344.反转字符串.ts
/*
 * @lc app=leetcode.cn id=344 lang=typescript
 *
 * [344] 反转字符串
 */

// @lc code=start
/**
 Do not return anything, modify s in-place instead.
 */
function reverseString(s: string[]): void {
    for (
        let left = 0, right = s.length - 1;
        left < right;
        left++, right--
    ) {
        s[left] = String.fromCharCode(s[left].charCodeAt(0) ^ s[right].charCodeAt(0))
        s[right] = String.fromCharCode(s[right].charCodeAt(0) ^ s[left].charCodeAt(0))
        s[left] = String.fromCharCode(s[left].charCodeAt(0) ^ s[right].charCodeAt(0))
    }
};
// @lc code=end

